pkg/http/core/kubernetes/deploy.go

```diff
@@ -65,19 +70,44 @@ func (d *deployManfest) Run() error {
 		if err != nil {
 			return err
 		}
+		application := d.dm.Moniker.App
 
-		name := u.GetName()
-
-		err = d.kc.AddSpinnakerAnnotations(u, d.dm.Moniker.App)
+		err = d.kc.AddSpinnakerAnnotations(u, application)
 		if err != nil {
 			return err
 		}
 
-		err = d.kc.AddSpinnakerLabels(u, d.dm.Moniker.App)
+		err = d.kc.AddSpinnakerLabels(u, application)
 		if err != nil {
 			return err
 		}
 
+		name := u.GetName()
+
+		if d.kc.IsVersioned(u) {
+			kind := strings.ToLower(u.GetKind())
+			namespace := u.GetNamespace()
+			lo := metav1.ListOptions{
+				LabelSelector:  kubernetes.LabelKubernetesName + "=" + application,
+				TimeoutSeconds: &listTimeout,
+			}
+
+			results, err := client.ListResourcesByKindAndNamespace(kind, namespace, lo)
+			if err != nil {
+				return err
+			}
+
+			currentVersion := d.kc.GetCurrentVersion(results, kind, name)
+			latestVersion := d.kc.IncrementVersion(currentVersion)
+
+			u.SetName(u.GetName() + "-" + latestVersion.Short)
```

Comment:
The name needs to be set using `latestVersion.Long`.

Reply:
done